### PR TITLE
Fix off by one in csc_premute and add test

### PR DESCRIPTION
**Pull request by KristofferC**
Fixes a simple mistake that happened when the original code got converted from C and adds a test. Also removed a faulty, unexported, untested `droptol` I added in another PR.

### CODE REVIEW

File: test/sparsedir/sparse.jl [hunk 1]
```diff
@@ -918,29 +917,36 @@ A = sparse(tril(rand(5,5)))
...
 perm = randperm(10)
 @test symperm(A,perm).colptr == [1,2,3,3,3,4,5,5,7,9,10]
 
-# droptol
+#csc_permute
+A = sprand(10,10,0.2)
+pinv = randperm(10)
+p = zeros(Int, 10)
+# Invert pinv
+for (i, j) in enumerate(pinv)
```
Comment (andreasnoack) [inline test/sparsedir/sparse.jl]: Couldn't you use `invperm` for this

File: test/sparsedir/sparse.jl [hunk 2]
```diff
@@ -918,29 +917,36 @@ A = sparse(tril(rand(5,5)))
...
+    p[j] = i
+end
+q = randperm(10)
+@test csc_permute(A, pinv, q) == full(A)[p, q]
```
Comment (andreasnoack) [inline test/sparsedir/sparse.jl]: Do you need to add the module here? `csc_permute` is not exported.

**Comment by KristofferC** [comment]
Fixed the stuff I was failing at.

This fails locally but I am pushing it anyway because it is failing in a place it shouldn't. If I put the test where I did now, the test [here](https://github.com/JuliaLang/julia/blob/89ca810594143c95df1857915a2b96b7a44dbe32/test/sparsedir/sparse.jl#L1078) fails.

If I put it in the bottom of the file, everything passes.

This seems to indicate that the test currently failing has some dependency on the random seed? @andreasnoack  any ideas?

**Comment by KristofferC** [comment]
The build log is too long so you have to download the raw log to see the test failure. Anyway here it is:

```julia
ERROR: LoadError: On worker 3:
LoadError("/tmp/julia/share/julia/test/sparse.jl",3,LoadError("/tmp/julia/share/julia/test/sparsedir/sparse.jl",1078,ErrorException("assertion failed: |cond(Ac,Inf) - cond(full(Ac),Inf)| <= 0.0001\n  cond(Ac,Inf) = 128.30458154354002\n  cond(full(Ac),Inf) = 125.08790073210642\n  difference = 3.216680811433605 > 0.0001")))
 in error at error.jl:22
 in test_approx_eq at test.jl:140
 in include_string at loading.jl:229
 in include_from_node1 at ./loading.jl:270
 in include_string at loading.jl:229
 in include_from_node1 at ./loading.jl:270
 in runtests at /tmp/julia/share/julia/test/testdefs.jl:179
 in anonymous at multi.jl:889
 in run_work_thunk at multi.jl:642
 in anonymous at task.jl:889
while loading /tmp/julia/share/julia/test/runtests.jl, in expression starting on line 13
```

**Comment by KristofferC** [comment]
@andreasnoack Should I just reorder the test to make this pass?

**Comment by KristofferC** [comment]
I did that now. Should be ready to merge.

**Comment by andreasnoack** [comment]
Thanks. I think it is ready now.

**Pull request merged**